modules/carto/src/api/layer-map.ts

```diff
@@ -46,6 +46,20 @@ const SCALE_FUNCS = {
 };
 export type SCALE_TYPE = keyof typeof SCALE_FUNCS;
 
+const RASTER_LAYER_TYPE = 'raster';
```

Comment:
I don't think you need the constants, instead do:

```ts
type LayerType = 'raster' | 'quadbin' | 'h3' | 'heatmapTile';
```

and then type `LAYER_FROM_LAYER_TYPE` as `Record<LayerType, Layer>`

Reply:
But that means repeating the same `'whatever'` everywhere, isn't it better to create a constant and use it 🤔 ?

Reply:
> then type LAYER_FROM_LAYER_TYPE as Record<LayerType, Layer>
Isn't that inferred automagically by Typescript?

Reply:
If that's the agreement, I'll follow it 👍 

On the other hand, I have to disagree with the "needless" of defining constants. Defining constants has many benefits:

* It's self documenting, because constants could have descriptive names, rather than "just values", for example, it's not the same `quadbin` in terms of layer representation than in terms of data (e.g. `QUADBIN_LAYER_TYPE` vs `QUADBIN_DATA_TYPE`)
* It's a single point of change, in case the LayerType gets modified, meaning with your proposal we'll have to modify several places
* We are avoiding human mistakes as typos, because maybe in some places it's typed as `string`, but not as `LayerType`
* Maybe I'm wrong, but modern JS engines optimize better constants than hard-coded literals
* Refactors are much easier with constants, because of the single point of change

Repeating the same value again and again in a codebase has many disadvantages, but if it's the agreement because that's the meaning of constants, avoid repeating the same again and again, and give it some sort of order 🤷

Reply:
In the case of "magic" values you may have a points, but 

```ts
const layerType: LayerType = 'raster';
```

is more readable as I actually know what the variable will hold. With a constant I have to go look it up.

As for the single point of change, we just update the type and fix the errors the compiler instantly throws up. 

Humans shouldn't type it as `string` 😉 it won't pass code review (as you can see!)